base/deprecated.jl

```diff
@@ -106,8 +106,12 @@ function depwarn(msg, funcsym; force::Bool=false)
         _module=begin
             bt = backtrace()
             frame, caller = firstcaller(bt, funcsym)
-            # TODO: Is it reasonable to attribute callers without linfo to Core?
-            caller.linfo isa Core.MethodInstance ? caller.linfo.def.module : Core
+            linfo = caller.linfo
+            if linfo isa Core.MethodInstance
+                linfo.def isa Module ? linfo.def : linfo.def.module
```

Comment:
```suggestion
                def = linfo.def
                def isa Module ? def : def.module
```

for maximum inferrability.